pkg/asset/machines/azure/azuremachines.go

```diff
@@ -0,0 +1,251 @@
+		}
+	}
+
+	providerID := fmt.Sprintf("%s-identity", clusterID)
+	var result []*asset.RuntimeFile
+	for idx := int64(0); idx < total; idx++ {
+		zone := mpool.Zones[int(idx)%len(mpool.Zones)]
+		azureMachine := &capz.AzureMachine{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				Kind:       "AzureMachine",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-%s-%d", clusterID, pool.Name, idx),
+				Labels: map[string]string{
+					"cluster.x-k8s.io/control-plane": "",
+					"cluster.x-k8s.io/cluster-name":  clusterID,
+				},
+			},
+			Spec: capz.AzureMachineSpec{
+				ProviderID:             ptr.To(providerID),
+				VMSize:                 mpool.InstanceType, // required
+				FailureDomain:          ptr.To(zone),
+				OSDisk:                 osDisk, // required
+				SSHPublicKey:           "",
+				AdditionalTags:         tags,
+				AdditionalCapabilities: additionalCapabilities,
+				AllocatePublicIP:       false,
+				EnableIPForwarding:     false,
+			},
+		}
+		*azureMachine.Spec.Image = *image
+		*azureMachine.Spec.AdditionalCapabilities = *additionalCapabilities
+		*azureMachine.Spec.SecurityProfile = *securityProfile
+
+		result = append(result, &asset.RuntimeFile{
+			File:   asset.File{Filename: fmt.Sprintf("10_inframachine_%s.yaml", azureMachine.Name)},
+			Object: azureMachine,
+		})
+
+		controlPlaneMachine := &capi.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: azureMachine.Name,
+				Labels: map[string]string{
+					"cluster.x-k8s.io/control-plane": "",
+				},
+			},
+			Spec: capi.MachineSpec{
+				ClusterName: clusterID,
+				Bootstrap: capi.Bootstrap{
+					DataSecretName: ptr.To(fmt.Sprintf("%s-%s", clusterID, role)),
+				},
+				InfrastructureRef: v1.ObjectReference{
+					APIVersion: "infrastructure.cluster.x-k8s.io/v1beta2",
+					Kind:       "AzureMachine",
+					Name:       azureMachine.Name,
+				},
+			},
+		}
+
+		result = append(result, &asset.RuntimeFile{
+			File:   asset.File{Filename: fmt.Sprintf("10_machine_%s.yaml", azureMachine.Name)},
+			Object: controlPlaneMachine,
+		})
+	}
+
+	bootstrapAzureMachine := &capz.AzureMachine{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+			Kind:       "AzureMachine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: capiutils.GenerateBoostrapMachineName(clusterID),
+			Labels: map[string]string{
+				"cluster.x-k8s.io/control-plane": "",
+				"install.openshift.io/bootstrap": "",
+				"cluster.x-k8s.io/cluster-name":  clusterID,
+			},
+		},
+		Spec: capz.AzureMachineSpec{
+			ProviderID:         ptr.To(providerID),
+			VMSize:             mpool.InstanceType, // required
+			Image:              image.DeepCopy(),
+			FailureDomain:      ptr.To(mpool.Zones[0]),
+			OSDisk:             osDisk, // required
+			SSHPublicKey:       "",
+			AdditionalTags:     tags,
+			AllocatePublicIP:   true,
+			EnableIPForwarding: false,
```

Comment:
nit: let's remove the fields that are using the default values, so drop `EnableIPForwarding` and `SSHPublicKey`